### PR TITLE
Consolidate sandboxing reproduction tests

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### Status
READY

### What does this PR accomplish?
- Groups all sandboxing reproductions into one context (one describe block with the same preparation steps for each test)
- Adds `USER_GROUPS` object and makes it available globally
- Abstracts repeated code into helper functions
- Includes refactor of `EE#535` reproduction (9b5ba270a80ce58ba6e6604f16b6b3423a89eac7)
- Includes repro for #13641 (4675c1cf581c2213f56e0ca63105d1e4389f6c53)

### Notes
- All tests should pass in CI

### CODE REVIEW

File: enterprise/frontend/test/metabase-enterprise/sandboxes/sandboxes.cy.spec.js [hunk 1]
```diff
@@ -914,3 +867,51 @@ describeWithToken("formatting > sandboxes", () => {
...
+      };
+
+      cy.log("**-- Update/save permissions --**");
+      cy.request("PUT", "/api/permissions/graph", {
+        groups,
+        revision,
+      });
+    },
+  );
+}
```
Comment (nemanjaglumac) [inline enterprise/frontend/test/metabase-enterprise/sandboxes/sandboxes.cy.spec.js:917]: This is probably the most important helper function for sandboxing tests.

I could've maybe made one for updating GTAP but it seems there's a bit more moving pieces there. This should be enough for now. Please let me know if you think different.

File: enterprise/frontend/test/metabase-enterprise/sandboxes/sandboxes.cy.spec.js [hunk 2]
```diff
@@ -914,3 +867,51 @@ describeWithToken("formatting > sandboxes", () => {
...
+} = {}) {
+  if (typeof schema !== "object") {
+    throw new Error("`schema` must be an object!");
+  }
+
+  cy.log("**-- Fetch permissions graph --**");
+  cy.request("GET", "/api/permissions/graph", {}).then(
+    ({ body: { groups, revision } }) => {
+      // This mutates the original `groups` object => we'll pass it next to the `PUT` request
+      groups[user_group] = {
```
Comment (paulrosenzweig) [inline enterprise/frontend/test/metabase-enterprise/sandboxes/sandboxes.cy.spec.js:902]: Maybe `groups[user_group][database_id] = {schemas:...` instead?

If we know that both those keys are defined, that change will prevent deleting the other databases if we add those down the line. Though, I'm not sure how the endpoint works so maybe not including some database ids for a group is fine anyways.

**Review by paulrosenzweig — APPROVED** [review]
This looks great!

File: enterprise/frontend/test/metabase-enterprise/sandboxes/sandboxes.cy.spec.js (hunk 2 above)
Comment (nemanjaglumac) [reply enterprise/frontend/test/metabase-enterprise/sandboxes/sandboxes.cy.spec.js:902]: I could've used something from `_underscore`, but I thought this was simple enough. As for the potential external/additional database ids, I am not sure how can we override those? The function accepts `database_id` parameter so it should be pretty straightforward and obvious which database schema we're updating, right?

If I'm not mistaken, the following scenario should be valid:
```javascript
updatePermissionsGraph({database_id: 1, schema: {...}})

// then immediately after that:
updatePermissionsGraph({database_id: 2, schema: {...}})
```

If I'm not mistaken it works cumulatively?

P.S. In any case, this function is (so far) localized to this file only. We can monitor it and add update it as needed.

Thank you for taking the time to review this.

Comment (paulrosenzweig) [reply enterprise/frontend/test/metabase-enterprise/sandboxes/sandboxes.cy.spec.js:902]: I'm assuming we need to post up the entire graph each time, but I haven't tested it. The way this is written now, if there are multiple dbs, the others will get dropped within `groups[user_group]`. 

`{[group_id]: {1: {schemas:...}, 2: {schemas:...}}}` => `{[group_id]: {1: {schemas:...}}}` if we set `groups[group_id] = {1:  {schemas:...}}`

Merging and waiting for any issues is a good call though 👍

**Pull request merged**